src/hotspot/share/opto/memnode.cpp

```diff
@@ -2095,7 +2095,7 @@ const Type* LoadNode::Value(PhaseGVN* phase) const {
       // Note:  When interfaces are reliable, we can narrow the interface
       // test to (klass != Serializable && klass != Cloneable).
       assert(Opcode() == Op_LoadI, "must load an int from _layout_helper");
-      jint min_size = Klass::instance_layout_helper(oopDesc::header_size(), false);
+      jint min_size = Klass::instance_layout_helper(checked_cast<jint>(heap_word_size(oopDesc::base_offset_in_bytes())), false);
```

Comment:
This is not the first time we do the whole `checked_cast<jint>(heap_word_size(oopDesc::base_offset_in_bytes())` dance. Maybe it should be a helper method straight in `oopDesc` then?